shell/platform/darwin/macos/framework/Source/FlutterBackingStoreDescriptor.h

```diff
@@ -0,0 +1,50 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Cocoa/Cocoa.h>
+#import <Metal/Metal.h>
+
+/**
+ * The type of the backing store, either Metal or OpenGL.
+ */
+typedef NS_ENUM(NSInteger, FlutterMacOSBackingStoreType) {
+  FlutterMacOSBackingStoreTypeMetal,
+  FlutterMacOSBackingStoreTypeOpenGL
+};
+
+/**
+ * A wrapper around backing store handles. On OpenGL, refers to the frame buffer id, and on Metal,
+ * holds the reference to the MTLTexture. Will only contain one of Metal or OpenGL backing store
+ * handles.
+ */
+@interface FlutterBackingStoreDescriptor : NSObject
+
+/**
+ * The type of the backing store, either Metal or OpenGL. See: FlutterMacOSBackingStoreType.
+ */
+@property(readonly, nonatomic) FlutterMacOSBackingStoreType backingStoreType;
+
+/**
+ * Initializes a backing store descriptor with the specified frame buffer id.
+ */
+- (nullable instancetype)initOpenGLDescriptorWithFBOId:(uint32_t)fboId;
```

Comment:
The naming of these init methods is not idiomatic; having what sounds like subclass names as part of the init isn't standard, and seems confusing to me. E.g., here it's not initializing an OpenGLDescriptor object, which it sort of sounds like. How about `initWithOpenGLFrameBufferID:` and `initWithMetalTexture:`